dataflow/src/ops/rewrite.rs

```diff
@@ -0,0 +1,240 @@
+use std::collections::{HashMap, HashSet};
+use prelude::*;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Rewrite {
+    src: IndexPair,
+    should_rewrite: IndexPair,
+
+    rw_col: usize,
+    value: DataType,
+
+    key: usize,
+}
+
+impl Rewrite {
+    pub fn new(src: NodeIndex, should_rewrite: NodeIndex, rw_col: usize, value: DataType, key: usize) -> Rewrite {
+        Rewrite {
+            src: src.into(),
+            should_rewrite: should_rewrite.into(),
+            rw_col: rw_col,
+            value: value,
+            key: key,
+        }
+    }
+
+    fn rewrite(&self, r: Vec<DataType>) -> Vec<DataType> {
+        let mut r = r.clone();
+        r[self.rw_col] = self.value.clone();
+        r
+    }
+
+    fn generate_row(&self, r: Vec<DataType>, positive: bool) -> Record {
+        if positive {
+            Record::Positive(r.clone())
+        } else {
+            Record::Negative(r.clone())
+        }
+    }
+}
+
+
+impl Ingredient for Rewrite {
+    fn take(&mut self) -> NodeOperator {
+        Clone::clone(self).into()
+    }
+
+    fn ancestors(&self) -> Vec<NodeIndex> {
+        vec![self.src.as_global(), self.should_rewrite.as_global()]
+    }
+
+    fn is_join(&self) -> bool {
+        true
+    }
+
+
+    fn must_replay_among(&self) -> Option<HashSet<NodeIndex>> {
+        Some(Some(self.src.as_global()).into_iter().collect())
```

Comment:
Just for future reference, you can use `std::iter::once(self.src.as_global())` if you want an iterator that yields just one value :) Not important to change here though.